### DISCUSSION TITLE
Oracle - disabled constraints are used anyway?

### DISCUSSION THREAD

**Discussion by FredericoCoelhoNunes** [sqlalchemy/sqlalchemy#6872] [Usage Questions]
Hi everyone, 

I have a table in my Oracle database, and its "unique" constrains look like this (apologies for the print screen, I can't copy paste text from my company's development environment to my local machine.....):

![image](https://user-images.githubusercontent.com/51457048/128860625-4a0faded-75b2-425c-8d65-eb4121b7be35.png)

It seems to me that the only unique constraints that are enables are NUDOSS and NULIGN. However, when I fetch the table using SQLALchemy, I get the following constraint:

![image](https://user-images.githubusercontent.com/51457048/128860814-936643c9-22ef-4af9-8dde-a10ea910ca39.png)

This is further confirmed whenI create the same table on MariaDb, using the metadata extracted from the table in Oracle:

![image](https://user-images.githubusercontent.com/51457048/128860964-6ea8a162-40ce-4ea6-9d1b-a6191885e9e8.png)

For some reason, SQLAlchemy is creating keys using the disabled constraints.

What am I missing here? Do I need to enable some parameter to prevent this from happening? For now I am manually removing these case-by-case as I find them during my database migration, but this isn't very robust and it can fail silently (adding one additional constraint that by chance works for now, but might not be intended).

Any insight would be greatly appreciated. Thank you!

**Comment by zzzeek** [comment]
hey there -

SQLAlchemy only emits CREATE TABLE and ADD CONSTRAINT statements, it knows nothing about the internals of how Oracle will be interpreting what is sent.   if you are seeing a CREATE TABLE that refers to also creating some kind of constraint, that would be in the source code that you are working with and you'd need to correct it in your local application.

To see what statements SQLAlchemy is emitting you would [enable logging](https://docs.sqlalchemy.org/en/14/core/engines.html#configuring-logging).    There's a one-to-one correspondence between DDL structures you see being created and constructs that are in your calling code, so that's where the problem would be.
